libpromises/evalfunction.c

```diff
@@ -238,7 +249,7 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             break;
 
         case RVAL_TYPE_CONTAINER:
-            convert = value;
+            convert = (JsonElement*) value;
```

Comment:
Done, thanks @jimis 

I hope we find a way to optimize this later. I added a TODO note.